source/mesh_refinement/compaction_length.cc

```diff
@@ -65,7 +70,7 @@ namespace aspect
                         ExcMessage("Need MeltOutputs from the material model for computing the melt properties."));
 
             // for each composition dof, check if the compaction length exceeds the cell size
-            for (unsigned int i=0; i<this->get_fe().base_element(this->introspection().base_elements.compositional_fields).dofs_per_cell; ++i)
+            for (unsigned int i=0; i<in.n_evaluation_points(); ++i)
```

Comment:
I would accept this commit separately if it's cumbersome to carry around.

Reply:
see #5816